### PR TITLE
Proposed restructuring for removal of fh-mbaas-express issue #2

### PR DESCRIPTION
**Pull request by david-martin**
Issue #2 

Changes:

* Rename from fh-sync-cloud to just fh-sync (having the 'cloud' in the name makes it sound less standalone)
* Remove the 'sync' sub-folder in lib, test & integration folders
* Remove the fh-mbaas-express coupling
* Remove all non sync code & tests
* Get all unit & integration tests working via `npm test`
* Add a sample server usage of the sync server `example/server.js`
* Reset the README.md file to start with standalone sync usage steps
* Remove unused dependencies from package.json that were carried over
from fh-mbaas-api

### CODE REVIEW

**Comment by david-martin** [comment]
@wtrocki @aidenkeating @wei-lee Can you review these changes (they are quiet verbose, so i highlighted the changes in the description).
This PR is aimed at completely decoupling fh-sync from any ties with fh-mbaas-* or RHMAP.
Through this decoupling, it should be possible to drop it into fh-mbaas-api as a required module instead.

fyi, i published a placeholder package to https://www.npmjs.com/package/fh-sync until we have a viable first release

File: example/server.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+var redisUrl = 'redis://127.0.0.1:6379';
+
+sync.api.connect(mongodbConnectionString, {}, redisUrl, function(){});
+
+sync.api.getEventEmitter().on('sync:ready', function() {
+  console.log('sync ready');
+
+  sync.api.init('myDataset', {
+    syncFrequency: 10 // seconds
+  }, function() {});
```
Comment (wtrocki) [inline example/server.js:13]: I like idea of single simplified server file instead of full app in #1 
Going to extend it a little bit after merge.

**Review by wtrocki — APPROVED** [review]
Reviewed and verified.

**Comment by david-martin** [comment]
@wtrocki thanks for merging. I've enabled the travis build for PRs now.
I've added you and some others to the https://www.npmjs.com/package/fh-sync package on npm too. Feel free to add any other collaborators I've missed

**Pull request merged**